hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioEventLoopGroup.java

```diff
@@ -335,4 +267,112 @@ public void onClose(Channel channel) {
             metricsRegistry.deregister(nioChannel.getWriter());
         }
     }
+
+    private class PublishAllTask implements Runnable {
+        @Override
+        public void run() {
+            for (NioChannel channel : channels) {
+                final NioChannelReader reader = channel.getReader();
+                NioThread inputThread = reader.getOwner();
+                if (inputThread != null) {
+                    inputThread.addTaskAndWakeup(new Runnable() {
+                        @Override
+                        public void run() {
+                            reader.publish();
+                        }
+                    });
+                }
+
+                final NioChannelWriter writer = channel.getWriter();
+                NioThread outputThread = writer.getOwner();
+                if (outputThread != null) {
+                    outputThread.addTaskAndWakeup(new Runnable() {
+                        @Override
+                        public void run() {
+                            writer.publish();
+                        }
+                    });
+                }
+            }
+        }
+    }
+
+    public static class Context {
+        private BackoffIdleStrategy idleStrategy;
+        private LoggingService loggingService;
+        private MetricsRegistry metricsRegistry;
+        private String threadNamePrefix = "hz";
+        private ChannelErrorHandler errorHandler;
+        private int inputThreadCount = 3;
+        private int outputThreadCount = 3;
```

Comment:
are these thread-counts intentionally hard-coded?

Reply:
Yes. But let me get them from the GroupProperties.

Reply:
I have change the default to 1 instead of 3. For both member/client the values will come from the configuration. For testing it either is set explicitly or.. it doesn't matter for the test and 1 is sufficient.